boa/src/builtins/bigint/mod.rs

```diff
@@ -131,18 +132,44 @@ impl BigInt {
         args: &[JsValue],
         context: &mut Context,
     ) -> Result<JsValue> {
-        let radix = if !args.is_empty() {
-            args[0].to_integer(context)? as i32
+        // 1. Let x be ? thisBigIntValue(this value).
+        let x = Self::this_bigint_value(this, context)?;
+
+        let radix = args.get(0).cloned().unwrap_or_default();
+
+        // 2. If radix is undefined, let radixMV be 10.
+        let radix_mv = if radix.is_undefined() {
+            // 5. If radixMV = 10, return ! ToString(x).
+            // Note: early return optimization.
+            return Ok(x.to_string().into());
+        // 3. Else, let radixMV be ? ToIntegerOrInfinity(radix).
         } else {
-            10
+            radix.to_integer_or_infinity(context)?
         };
-        if !(2..=36).contains(&radix) {
+
+        // 4. If radixMV < 2 or radixMV > 36, throw a RangeError exception.
+        let radix_mv = if let IntegerOrInfinity::Integer(i) = radix_mv {
+            if !(2..=36).contains(&i) {
+                return context.throw_range_error(
+                    "radix must be an integer at least 2 and no greater than 36",
+                );
+            }
+            i
+        } else {
             return context
                 .throw_range_error("radix must be an integer at least 2 and no greater than 36");
+        };
```

Comment:
Maybe a `match` with a `if !(2..=36).contains(&i)` guard would be better to avoid repeating the same error twice

Reply:
I've been thinking on how to not duplicate the error for like 3 minutes 🤦‍♂️. Thanks :)

Reply:
No problem. Glad I could help :D